app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala

```diff
@@ -139,8 +143,14 @@ class IfConnector @Inject()(
       auditHelper.auditIfApiFailure(correlationId, matchId, request, requestUrl, msg)
       msg.contains("NO_DATA_FOUND") match {
         case true =>
-          logger.warn("Integration Framework NotFoundException encountered")
-          Future.failed(new NotFoundException(msg))
+          if (requestUrl.contains("counts"))
```

Comment:
I dont really like the placement of this, why should the IF connector simply used for making calls to IF know about all the different URLs. I'd prefer to add this recoery code to the individuals services as they are already specific to each endpoint so makes sense to have these moved to each service. Also stops this growing as future devs add more and more endpoints.

Reply:
@parkeradam  - The IF connector already knows about the different URLs as they are coded into the IF connector. The recover method is generic and applies to all endpoints. The empty objects being returned are IF objects and created via the IF calls. I will however explore alternative(s) and see what that might look like in code.

Reply:
`requestUrl` is the IF request.

Reply:
My bad, didnt see in the context of the whole file.